manifests/augeas.pp

```diff
@@ -9,7 +9,7 @@
     create_resources(augeas, $config, $defaults)
   }
   else {
-    $hiera_config = hiera_hash('system::augeas', undef)
+    $hiera_config = lookup({'name' => 'system::augeas', 'value_type' => Variant[Undef, Hash], 'merge' => { 'strategy' => 'deep', }, 'default_value' => undef })
```

Comment:
I thought about this for some time. Why do we even do `lookup()` calls? automatic parameter lookup would be way easier? https://puppet.com/docs/puppet/latest/hiera_automatic.html#class-parameters